creator-node/src/AsyncProcessingQueue.js

```diff
@@ -39,22 +47,49 @@ class AsyncProcessingQueue {
       }
     })
 
+    this.libs = null
+
     this.queue.process(MAX_CONCURRENCY, async (job, done) => {
       const { logContext, task } = job.data
 
       const func = this.getFn(task)
 
-      try {
-        const response = await this.monitorProgress(task, func, job.data)
-        done(null, { response })
-      } catch (e) {
-        this.logError(
-          `Could not process taskType=${task} uuid=${
-            logContext.requestID
-          }: ${e.toString()}`,
-          logContext
-        )
-        done(e.toString())
+      if (task === PROCESS_NAMES.transcodeHandOff) {
+        const { transcodeFilePath, segmentFileNames, sp } =
+          await this.monitorProgress(task, transcodeHandOff, job.data)
+
+        if (!transcodeFilePath || !segmentFileNames) {
+          this.logStatus(
+            'Failed to hand off transcode. Retrying upload to current node..'
+          )
+          await this.addTrackContentUploadTask({
+            logContext,
+            req: job.data.req
+          })
+          done(null, {})
+        } else {
+          this.logStatus(
+            `Succesfully handed off transcoding and segmenting to sp=${sp}. Wrapping up remainder of track association..`
+          )
+          await this.addProcessTranscodeAndSegmentTask({
+            logContext,
+            req: { ...job.data.req, transcodeFilePath, segmentFileNames }
+          })
+          done(null, { response: { transcodeFilePath, segmentFileNames } })
+        }
+      } else {
+        try {
+          const response = await this.monitorProgress(task, func, job.data)
```

Comment:
is this the retry? `monitorProgress` doesn't make it super clear imo that it's handing off the transcode

Reply:
monitorProgress is a fn that takes in a task fn as a param, and adds the state of the job and any job responses to redis. it's not the retry, but a general fn that processes certain tasks depending on the task name